src/Request/ConsoleRequest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Bugsnag\Request;
+
+class ConsoleRequest implements RequestInterface
+{
+    /**
+     * The unformated console command.
+     *
+     * @var array
+     */
+    protected $command;
+
+    /**
+     * Create a new console request instance.
+     *
+     * @param array     $commandArray An array of the command line input
+     *
+     * @return void
+     */
+    public function __construct(array $commandArray)
+    {
+        $this->command = $commandArray;
+    }
+
+    /**
+     * Are we currently processing a request?
+     *
+     * @return bool
+     */
+    public function isRequest()
+    {
+        return true;
+    }
+
+    /**
+     * Get the session data.
+     *
+     * @return array
+     */
+    public function getSession()
+    {
+        return [];
+    }
+
+    /**
+     * Get the cookies.
+     *
+     * @return array
+     */
+    public function getCookies()
+    {
+        return [];
+    }
+
+    /**
+     * Get the request formatted as meta data.
+     *
+     * @return array
+     */
+    public function getMetaData()
+    {
+        if (is_array($this->command)) {
+            $commandString = implode(' ', $this->command);
+            $primaryCommand = $this->command[0];
+            $arguments = [];
+            $options = [];
+            foreach (array_slice($this->command, 1) as $arg) {
+                $arg[0] == '-' ? $options[] = $arg : $arguments[] = $arg;
+            }
+            $data = [
+                'Input' => $commandString,
+                'Command' => $primaryCommand,
+                'Arguments' => $arguments,
+                'Options' => $options,
+            ];
+        } else {
+            $data = $this->command;
+        }
+
+        return ['console' => $data];
+    }
+
+    /**
+     * Get the request context.
+     *
+     * @return string|null
+     */
+    public function getContext()
+    {
+    }
```

Comment:
`return implode(' ', $this->command)` perhaps would be a good context, or perhaps, limit this to only the first upto 2 parts of the array?

Reply:
The first two parts would be good, running this with `artisan` commands as well as plain `php` produces useful results, moreso than the default which is the top line of the stacktrace.

Reply:
This code will return at most the first 2 parts of the command, concatenated together with a space. If the array is smaller than 2 elements, it is tolerant to that, and will just return the singleton array, or the empty array.

```php
return implode(' ', array_slice($this->command, 0, 2));
```